### DISCUSSION TITLE
MQTT Broker

### DISCUSSION THREAD

**Discussion by gdg316** [tsightler/ring-mqtt#464] [Support Q&A]
Hey does anyone know why I am seeing this ?? Should Debugging be enabled for MQTT Broker

Client <unknown> closed its connection
2022-09-12 06:28:10: Client <unknown> closed its connection. 
Client Unknown Closed its Connection

So my question is I have Homeassistant OS on Raspberry PI 4
And I followed directions for Mosquito Broker which said create a Username and Password in The Frontend under users and it does not need to be Administrator.
Then I reconfigured MQTT with those credentials and I had communication restored to Ring MQTT. It stated that username could not be Homeassistant or add on.
Should I reinstall MQTT Broker ??
And then reinstall Ring-MQTT ??
A new error says Client Unknown Closed it's Connection.And should Debugging be on for MQTT Broker ??
Ring-MQTT Configuration I never altered or made changes.
So I see auto username and auto password in ring MQTT URL
But Auto hostname ? And my too assume my local HTTP link from web browser?? Or from Hassio itself ?

**Comment by joeu1998** [comment]
I’m also wondering how I should be configuring the Ring-MQTT configuration. 

I haven’t changed anything, although it says in the installation guide that I should review the configuration before starting the container for the first time. 

I’ve reviewed the configuration info on this repo, but don’t understand what I should be setting the URL to, same as OP, nor the username / password (or if I even need to?) - is this the user I’ve already creating in HA Users, for the MQTT broker, or is this a new user / password? What should the Ring-MQTT ‘mqtt_url’ be set as? 

Thank you in advance!

**Reply by tsightler** [reply]
If you are using the addon and you've set up the Mosquito addon and Mosquitto HA integration using the default setup as described in the HA documentation then, in general, you won't need to change anything as the addon support automatic discovery of the MQTT service parameters.  If you've customized your MQTT setup, then you'll need to set the MQTT URL as appropriate for connection to your MQTT broker.  It's really not that complex and an example is provided in the configuration documentation so it's hard to understand what is not clear since you say you have reviewed this information.

**Reply by joeu1998** [reply]
Thanks @tsightler for responding so quickly, I have indeed set up the MQTT addon and integration completely as default, with a new user created in HA Users, which is the only custom info I plugged into the MQTT integration. I couldn’t find step by step documentation on HA’s site for correct MQTT installation, so I followed a YouTube video - but left everything as default, aside from that username / password I mentioned. 

the ports on the mqtt addon are default, and I didn’t customise the config here, either. It just has:

active: false
folder: mosquitto 

So does this mean for the mqtt_url in the Ring-mqtt addon, it’s best to just leave everything as default? For me this is currently: 

mqtt://auto_username:auto_password@auto_hostname

Apologies in advance if these are ridiculous questions - everything is working with the live feed / snapshot image, but I wanted to make sure I’d set everything up correctly and securely before leaving it running.

**Reply by tsightler** [reply]
Yes, "auto" really means auto, the addon will automatically detect these values using the service discovery API.  If everything is working then clearly there is nothing you need to do.  Perhaps I need to make the documentation more clear that, when using the addon installation method, the default behavior is to automatically detect the MQTT service settings and you only need to change things if you need to override the default behavior (for example, when using an external MQTT broker).
